### ISSUE TITLE
[tabular] Add log-scaling to regression for appropriate metrics

### ISSUE THREAD

**Issue by Innixma** [autogluon/autogluon#4218]
In regression tasks for metrics such as RMLSE (root mean squared log error), it is beneficial to first log scale the ground truth before training, and then fit with RMSE. At predict time, we can inverse the log transform to get the corrected predictions.

This is because passing RMLSE directly for the purposes of early stopping fails to align with the way the model's loss function is being optimized. But using RMSE lets it be much more closely aligned.

This process should also be done for all other cases where log transforms occur (and potentially other transforms).

### Internal Handling

One candidate idea is add extra logic to `Scorer`. This logic would include `.transform(y)` and `.inverse_transform(y)`. We can then refer to these transforms when scoring predictions and returning predictions to the user.

One downside of this approach is added code complexity. If we ever forget to do the transform / inverse transform at any point, the scores would become incorrect and performance would severely degrade. We can ward against this with careful unit testing and benchmarking of the metrics this applies to.

Maybe this logic can also be tied to the Models, ensuring the transform and inverse transforms happen in the `predict`, `predict_proba` and `score` methods, among others. We will need to ensure that this doesn't happen multiple times in the cases of bagged ensembles or nested models.

We could use an `ag.metric_transform` ag_args_fit argument to enable or disable the transforms in the models.

### External Handling

Another option is to have TabularPredictor or Trainer handle the transform. TabularPredictor is less ideal because all the scores reported in logging and leaderboard would be incorrect, but it would be the easiest to make work without bugs. For Trainer, we would be able to make the logging and leaderboard correct, but things might become odd with model-specific logging.

**Comment by albertcthomas** [comment]
related to this but it's more a user question: for the RMSLE, I can for instance define a custom metric implementing it and pass it as the `eval_metric`, but some models will optimize e.g. the RMSE (because the RMSLE is not recognized by autogluon and autogluon does not know how to deal with it rather than using the default metric of the model) and this can be suboptimal. The custom eval_metric will only be used for say building the best ensemble etc...?

**Comment by albertcthomas** [comment]
thanks the detailed answer :)
